nbgrader/apps/baseapp.py

```diff
@@ -190,6 +190,11 @@ def _load_config(self, cfg, **kwargs):
             cfg.Exchange.course_id = cfg.NbGrader.course_id
```

Comment:
I took a look myself and I think the only thing that might be missing is just to update this check fo the course_id being in NbGrader---I think you can change this to `cfg.CourseDirectory.course_id = cfg.NbGrader.course_id` (and then also update the warning just above as well)

Reply:
Good catch... I forgot about that one! Pushing the change now!